ambry-messageformat/src/test/java/com/github/ambry/messageformat/MessageFormatInputStreamTest.java

```diff
@@ -70,22 +70,22 @@ public void after() {
   @Test
   public void messageFormatPutRecordsTest() throws IOException, MessageFormatException {
     messageFormatPutRecordsTest(MessageFormatRecord.Blob_Version_V1, BlobType.DataBlob,
-        MessageFormatRecord.Message_Header_Version_V1);
-    messageFormatPutRecordsTest(MessageFormatRecord.Blob_Version_V2, BlobType.DataBlob,
-        MessageFormatRecord.Message_Header_Version_V1);
-    messageFormatPutRecordsTest(MessageFormatRecord.Blob_Version_V2, BlobType.MetadataBlob,
-        MessageFormatRecord.Message_Header_Version_V1);
-    messageFormatPutRecordsTest(MessageFormatRecord.Blob_Version_V2, BlobType.DataBlob,
-        MessageFormatRecord.Message_Header_Version_V2);
-    messageFormatPutRecordsTest(MessageFormatRecord.Blob_Version_V2, BlobType.MetadataBlob,
-        MessageFormatRecord.Message_Header_Version_V2);
-    messageFormatPutRecordsTest(MessageFormatRecord.Blob_Version_V2, BlobType.DataBlob,
-        MessageFormatRecord.Message_Header_Version_V3);
-    messageFormatPutRecordsTest(MessageFormatRecord.Blob_Version_V2, BlobType.MetadataBlob,
-        MessageFormatRecord.Message_Header_Version_V3);
+        MessageFormatRecord.Message_Header_Version_V1, false);
```

Comment:
If we support invoking serialization using old versions, it will make the serialization unnecessarily complex.  First, we should never serialize using old versions because it could lead to data lost.  For example, the chunk is compressed and compression flag is introduced in V3.  If we serialize using V2, the compression flag will be lost and will result in data lost.  So we should always serialize using the latest version.

Coming back to the need to test serializing using older version.  If we add a parameter deep in the serialization, we will need to pass this serialization version through many layers of code.  Again, it's dangerous to pass old version so if someone mistakenly pass in an older version, it can get deployed without failing any tests.

Instead, I suggest we keep these tests serializing using the latest version V3, but have different unit test to make sure V2 specifically to make sure their serializer and de-serializer are compatible.